pilot/pkg/xds/ads_test.go

```diff
@@ -979,3 +984,47 @@ func TestPushQueueLeak(t *testing.T) {
 	ds.Discovery.AdsPushAll(&model.PushRequest{Push: ds.PushContext()})
 	p.Cleanup()
 }
+
+func TestDistribution(t *testing.T) {
+	xds.ResetConnectionNumberForTest()
+	s := xdsfake.NewFakeDiscoveryServer(t, xdsfake.FakeOptions{})
+	expectNonce := func(nonce, ty string) {
+		t.Helper()
+		assert.EventuallyEqual(t, func() string {
+			return s.Discovery.StatusReporter.QueryLastNonce("test.default-1", ty)
+		}, nonce[:xds.VersionLen])
+	}
+
+	ledger := ledger.Make(time.Minute)
+	ledger.Put("key", "value") // If there is no config, ledger would be empty
+	s.Env().SetLedger(ledger)
+	reporter := &distribution.Reporter{
+		UpdateInterval: features.StatusUpdateInterval,
+	}
+	reporter.Init(s.Env().GetLedger(), test.NewStop(t))
+	s.Discovery.StatusReporter = reporter
+
+	ads := s.ConnectADS().WithType(v3.ClusterType)
+	// Subscribe to clusters
+	res1 := ads.RequestResponseAck(t, &discovery.DiscoveryRequest{ResourceNames: []string{"fake-cluster"}})
+	expectNonce(res1.Nonce, v3.ClusterType)
```

Comment:
```suggestion
	expectNonce(res1.Nonce, v3.ClusterType)
		// Types we are not subscribed to are updated immediately
	expectNonce(res.Nonce, v3.RouteType)
```

Sorry, I forgot this case in my original PR, and it fails on this PR. This is the behavior I have been trying to get you to retain by not removing the report in the push flow.

Reply:
Yes, i believe this is the right behavior.

Reply:
No, the suggestion addition to the test is the correct and current behavior. Sorry I feel like we are not communicating well here but I don't know any other ways to express this..

Reply:
Added this check, i understand your comment now